southbound/impl/src/main/java/org/opendaylight/ansible/southbound/AnsibleCommandServiceImpl.java

```diff
@@ -8,30 +8,150 @@
+    private final DataBroker dataBroker;
 
     @Inject
-    public AnsibleCommandServiceImpl() {
-        LOG.info("constructor");
+    public AnsibleCommandServiceImpl(@OsgiService final DataBroker dataBroker) {
+        this.dataBroker = dataBroker;
     }
 
     @Override
     public ListenableFuture<RpcResult<RunAnsibleCommandOutput>> runAnsibleCommand(RunAnsibleCommandInput input) {
-        LOG.info("We made it!");
-        return RpcResultBuilder.success(new RunAnsibleCommandOutputBuilder().setStatus("OK").build()).buildFuture();
+        Uuid uuid;
+        String status;
+        try {
+            uuid = runAnsible(input.getHost(), input.getDirectory(), input.getFile());
+            status = "INPROGRESS";
+        } catch (ManagedProcessException e) {
+            status = "FAILED";
+            uuid = null;
+        }
+        return RpcResultBuilder.success(new RunAnsibleCommandOutputBuilder().setStatus(status).setUuid(uuid).build())
+                .buildFuture();
+    }
+
+    private Uuid runAnsible(String host, String dir, String file) throws ManagedProcessException {
+        Uuid uuid = new Uuid(UUID.randomUUID().toString());
+        LOG.info("Executing Ansible, new uuid for command is: " + uuid);
+        ManagedProcessBuilder ar = new ManagedProcessBuilder("ansible-runner").addArgument("-j").addArgument
+                ("--hosts").addArgument(host).addArgument("-p").addArgument(file);
+        ar.addArgument("run").addArgument(dir);
+        ar.setProcessListener(new AnsibleProcessListener(this, uuid));
+        ManagedProcess p = ar.build();
+        processMap.put(uuid, p);
+        LOG.info("Starting Ansible process");
+        try {
+            p.start();
+            LOG.info("Ansible Process is alive: {}", Boolean.toString(p.isAlive()));
+        } catch (ManagedProcessException e) {
+            LOG.warn("Process exited with error code: " + p.getProcLongName());
+        }
+        return uuid;
+    }
+
+    public void parseAnsibleResult(Uuid uuid) throws AnsibleCommandException {
+        ManagedProcess p = getProcess(uuid);
+        if (p == null) {
+            throw new AnsibleCommandException("Unable to find process for uuid" + uuid.toString());
+        }
+        parseAnsibleResult(p, uuid);
+    }
+
+    public void parseAnsibleResult(ManagedProcess p, Uuid uuid) {
+        String output = p.getConsole();
+        String result;
+        String failedEventOutput = null;
+        LOG.info("Ansible process complete: {}", output);
+        try {
+            LOG.info("Parsing json string into Event List");
+            AnsibleEventList el = new AnsibleEventList(parseAnsibleOutput(output));
+            AnsibleEvent lastEvent = el.getLastEvent();
+            LOG.info("Stdout of last event is {}", lastEvent.getStdout());
+            if (el.AnsiblePassed()) {
+                LOG.info("Ansible Passed for " + p.getProcLongName());
+                result = "COMPLETE";
```

Comment:
Guess we should add constants for these.
Complete or Success?

Reply:
yeah i'll make them into constants

Reply:
will use enum in the yang to define valid results